ecs_pod_creator.go

```diff
@@ -1224,6 +1259,163 @@ func MergeECSPodExecutionOptions(opts ...ECSPodExecutionOptions) ECSPodExecution
 	return merged
 }
 
+// ECSOverridePodDefinitionOptions are options that can be specified when
+// starting a pod that override those in the pod's definition.
+type ECSOverridePodDefinitionOptions struct {
```

Comment:
Note that this could have been implemented by embedding this struct in the other structs (such as `ECSPodDefinitionOptions`) to reduce a bit of copy-pasted fields. Unfortunately, this results in other problems because the overridden field's struct layout and methods are at odds with their outer struct's desired behavior. For example, we still have to duplicate setter methods, and validation of the overridden fields depends on whether the field is being used for `RunTask` or `RegisterTaskDefinition`. I opted to not go the embedded struct route since the only thing that `RunTask` and `RegisterTaskDefinition` really share are the field names.

Reply:
Can we add a quick comment mentioning this so nobody in the future tries to "fix" it?

Reply:
Good idea, done.